### PR TITLE
Fixes following tests of NIDM-Results 1.0.0 with FSL exporter

### PR DESCRIPTION
**Pull request by TomMaullin**
This PR fixes all bugs that currently don't allow this NIDMResults branch to be compatible with the current NIDMResults FSL export.

Current fixes implemented:

- Small fix for 'type' variable in the NeuroimagingSoftware class.
- The rpv_map variable is now optional for the ModelFitting class.
- Fixing of instantiation of ContrastVariance object in ContrastStdError class.
- Adding of export_dir as an input to the ContrastStdError class as it is needed for the variance files path.
- Preventing errors caused by non-existent fields in 'exporter.py'.
- Changed the initialization of 'prepend_path' in 'export.py' from None to an empty string to avoid type errors.
- Changing of .tolist() for nifti dimensions as numpy int64s cannot be json serialized.  
- Changing 'self.contrast_name' to 'contrast_name' in the init function of InferenceActivity (and similar replacements in DisplayMaskMap, HeightThreshold and ContrastMeanExplainedSquaresMap).
- A drift model is only added now if it is available (Before the code was trying to output a 'None' drift model).
- Renaming of 'format' to 'fmt'.
- Fixing of indentation/logic in ContrastStdErrorMap.

### CODE REVIEW

**Comment by cmaumet** [comment]
Thanks @TomMaullin. Just for reference: this PR is made in conjunction with https://github.com/incf-nidash/nidmresults-fsl/pull/124 that updates the FSL exporter.

**Comment by TomMaullin** [comment]
Hi @cmaumet ,

Apologies for the delay (I forgot these tests take an hour to run on Travis!!), this branch should now be ready for review. Please let me know what you think of the changes I've made and whether they all seem in keeping with what you had planned for this branch!

**Comment by cmaumet** [comment]
@TomMaullin: Thanks for all your help on this!

I've added a couple of edits to this PR to solve the remaining issues in the FSL exporter tests:
 - Added an argument that can specify which attributes should be ignored when doing the comparison (for the FSL exporter test, we ignore the NIDM exporter version and the export time that can vary)
 - "Supra-threshold cluster" was used in the label of Cluster entities for NIDM version 1.0.0 and 1.1.0 while it should have been "Significant Cluster" (use for "supra-threshold" was introduced in NIDM 1.2.0)
 - Effects degrees of freedom for T and Z statistics set to 1 (as before)

File: nidmresults/objects/generic.py [hunk 1]
```diff
@@ -161,6 +161,8 @@ def __init__(self, coordinate_system, nifti_file=None, vox_to_world=None,
 
         self.voxel_to_world = vox_to_world
         self.voxel_size = vox_size
+        if type(dimensions[1])==np.int64:
```
Comment (cmaumet) [inline nidmresults/objects/generic.py]: @TomMaullin: do you remember why this statement was needed and why overwriting `dimensions`? `int64` seems quite restrictive and I would like to check if there is an alternative solution. 

Apart from that, everything looks good to me and I think we should merge once the tests are okay at https://github.com/incf-nidash/nidmresults-fsl/pull/124.

Comment (TomMaullin) [reply nidmresults/objects/generic.py]: I had a lot of trouble with this... I can't remember which way around this was but there was some discrepancy in the type of `dimensions`.

I can't remember which way around it was but, `dimensions` is either read in as a `int` `tuple` or an `np.int64` `np.array` depending on whether you are running the `nidmresults` or `nidmresults-fsl` tests. If I remember correctly, one of these data types came straight from a nibabel function and the other came straight from a JSON reader so I had no choice but to try and account for both.

This if statement checking whether it was `np.int64` is here as even if I converted the `tuple` to a `np.array` (or vice versa) the tests would still error as the type was `np.int64` and not normal `int` or (vice versa). If the data is a numpy type integer the only way I could account for this was by converting them to a normal integer type using the function `.item()` as I have on the line below.

**Comment by TomMaullin** [comment]
Hi @cmaumet , these changes make sense to me and the tests all now pass. Are you happy for us to merge this?

File: nidmresults/objects/generic.py (hunk 1 above)
Comment (cmaumet) [reply nidmresults/objects/generic.py]: Okay... Thanks! Let me try this out and see if I can do anything about it.

**Comment by cmaumet** [comment]
@TomMaullin: I've removed the conditional statement discussed in https://github.com/cmaumet/nidmresults/pull/1#discussion_r190833782 by making sure that `dimensions` is converted to a numpy array (when extracting the dimensions from a NIfTI image as is done with the FSL exporter). 

Let's make sure that the tests pass both here and at https://github.com/incf-nidash/nidmresults-fsl/pull/124 and then we can merge.

**Comment by cmaumet** [comment]
Let's merge! Thanks @TomMaullin!

**Pull request merged**